src/exponential.jl

```diff
@@ -238,7 +238,7 @@ function scale_and_square(A::IntervalMatrix{T}, l::Integer, t, p;
         end
     end
 
-    A_scaled = A / (interval(T(2))^l)
+    A_scaled = A / (Interval(T(2), T(2))^l)
```

Comment:
why not
```suggestion
    A_scaled = A / 2^l
```
?
or `scale(A_scaled, 1/2^l)`, using [scale](https://juliareach.github.io/IntervalMatrices.jl/latest/lib/methods/#IntervalMatrices.scale).

Reply:
I thought about that, but then I am not sure whether the computations include rounding errors.

Reply:
Yes, they consider rounding, see https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/master/src/intervals/arithmetic.jl#L150

Reply:
That is good, but the evaluation of `2^l` may already contain rounding errors. In this particular case (a power or two) I guess this is not the case for reasonably small `l`. So I am okay with changing it.

Reply:
True, `2^l` spits integers for integer `l` so it doesn't contain rounding errors; for "large" exponents recall that it overflows, 

```julia
julia> f(k) = 2^k;

julia> typeof(f(5))
Int64

julia> f(63)
-9223372036854775808

julia> 2^big(63)
9223372036854775808

julia> f(64)
0

julia> 2^big(64)
18446744073709551616
```

Reply:
Should we assert that `l < 63`? (or use BigInts)

Reply:
Or we just use what I had originally: do the computation in `Float64`.

Reply:
> Or we just use what I had originally: do the computation in Float64.

Ok.

Reply:
Done. I also changed the `2^l` in the validation to a float computation to prevent an overflow.